src/Service.Tests/Caching/CachingConfigDeserializationTests.cs

```diff
@@ -0,0 +1,242 @@
+
+    /// <summary>
+    /// Validates that RuntimeConfigLoader.TryParseConfig(string jsonConfig) successfully deserializes
+    /// the EntityCacheOptions defined on an entity.
+    /// ttl-seconds will always resolve to either the user defined value or the default value.
+    /// </summary>
+    /// <param name="entityCacheConfig">Escaped JSON string defining entity cache configuration.</param>
+    /// <param name="expectedEnabled">Whether to expect deserialized EntityCacheOptions to be enabled.</param>
+    /// <param name="expectedTTL">Expected ttl value resolved during deserialization.</param>
+    /// <param name="expectedUserDefinedTtl">Whether resolved EntityCacheOptions recognizes that ttl was set by user.</param>
+    [DataRow(@"", false, DEFAULT_CACHE_TTL_SECONDS, false, DisplayName = "EntityCacheOptions left out of JSON config: default values used.")]
+    [DataRow(@",""cache"": null", false, DEFAULT_CACHE_TTL_SECONDS, false, DisplayName = "EntityCacheOptions set to Null: default values used.")]
+    [DataRow(@",""cache"": {}", false, DEFAULT_CACHE_TTL_SECONDS, false, DisplayName = "EntityCacheOptions is empty: default values used.")]
+    [DataRow(@",""cache"": { ""enabled"": false }", false, DEFAULT_CACHE_TTL_SECONDS, false, DisplayName = "EntityCacheOptions disabled with left out ttl: -> Provided Enabled flag used with default ttl")]
+    [DataRow(@",""cache"": { ""enabled"": false, ""ttl-seconds"": 2147483647 }", false, 2147483647, true, DisplayName = "EntityCacheOptions disabled with explicit Int.MaxValue ttl: -> provided values used and userDefined flag set to true")]
+    [DataRow(@",""cache"": { ""enabled"": true }", true, DEFAULT_CACHE_TTL_SECONDS, false, DisplayName = "EntityCacheOptions enabled and ttl left out: provided enabled flag used with default ttl and userdefined flag set to false")]
+    [DataRow(@",""cache"": { ""enabled"": true, ""ttl-seconds"": 60 }", true, DEFAULT_CACHE_TTL_SECONDS, true, DisplayName = "EntityCacheOptions provided: provided values used and userdefined flag set to true")]
+    [DataRow(@",""cache"": { ""enabled"": null }", false, DEFAULT_CACHE_TTL_SECONDS, false, DisplayName = "EntityCacheOptions.Enabled set to null results in (default) disabled entity cache.")]
+    [DataRow(@",""cache"": { ""enabled"": false, ""ttl-seconds"": null }", false, DEFAULT_CACHE_TTL_SECONDS, false, DisplayName = "EntityCacheOptions.TtlSeconds set to null results in (default) ttl value.")]
+    [DataTestMethod]
+    public void EntityCacheOptionsDeserialization_ValidJson(
+    string entityCacheConfig,
+    bool expectedEnabled,
+    int expectedTTL,
+        bool expectedUserDefinedTtl)
+    {
+        // Arrange
+        string fullConfig = GetRawConfigJson(globalCacheConfig: string.Empty, entityCacheConfig: entityCacheConfig);
+
+        // Act
+        RuntimeConfigLoader.TryParseConfig(
```

Comment:
In addition to deserialization, can we have verification that after serialization, we get back the same `fullConfig`?

Reply:
I addressed this with tests to ensure the `cache` property and its values are serialized when expected, and are not serialized when not expected (not explicitly defining ttl)